.devcontainer/install-dependencies.sh

```diff
@@ -49,62 +111,95 @@ export GOPATH=$TMPDIR
 export GOCACHE=$TMPDIR/cache
 export GO111MODULE=on
 
-echo "Installing Go tools…"
+write-verbose "Installing Go tools…"
 
 # go tools for vscode are preinstalled by base image (see first comment in Dockerfile)
-go install k8s.io/code-generator/cmd/conversion-gen@v0.22.2 
-go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0 
-go install sigs.k8s.io/kind@v0.11.1 
-go install sigs.k8s.io/kustomize/kustomize/v4@v4.5.2 
+
+# go-install() is a helper function to trigger `go install` 
+# if a command is missing OR if we're using --force
+go-install() {
+    write-verbose "Checking for $1"
+    if [ "$FORCE" == true ] || [ ! -f "$GOBIN/$1" ]; then 
+        write-info "Installing $1"
+        shift # Discard the command name so we can pass the remaining arguments to GO
+        go install $@
+    fi
+}
+
+go-install conversion-gen k8s.io/code-generator/cmd/conversion-gen@v0.22.2 
+go-install controller-gen sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0 
+go-install kind sigs.k8s.io/kind@v0.11.1 
+go-install kustomize sigs.k8s.io/kustomize/kustomize/v4@v4.5.2 
 
 # for docs site
-go install -tags extended github.com/gohugoio/hugo@v0.88.1
-go install github.com/wjdp/htmltest@v0.15.0
+go-install hugo -tags extended github.com/gohugoio/hugo@v0.88.1
+go-install htmltest github.com/wjdp/htmltest@v0.15.0
 
 # for api docs 
 # TODO: Replace this with the new release tag.
-go install github.com/ahmetb/gen-crd-api-reference-docs@v0.3.1-0.20220223025230-af7c5e0048a3
-
-if [ "$1" != "devcontainer" ]; then 
-    echo "Installing golangci-lint…"
-    # golangci-lint is provided by base image if in devcontainer
-    # this command copied from there
-    curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "$TOOL_DEST" v1.45.2 2>&1
+go-install gen-crd-api-reference-docs github.com/ahmetb/gen-crd-api-reference-docs@v0.3.1-0.20220223025230-af7c5e0048a3
+
+# Install golangci-lint
+if [ "$DEVCONTAINER" != true ]; then 
+    write-verbose "Checking for golangci-lint"
+    if [ "$FORCE" == true ] || [ ! -f "$TOOL_DEST/golangci-lint" ]; then 
+        write-info "Installing golangci-lint"
+        # golangci-lint is provided by base image if in devcontainer
+        # this command copied from there
+        curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "$TOOL_DEST" v1.45.2 2>&1
+    fi
 fi
 
-# Install go-task (task runner)
-echo "Installing go-task…"
-curl -sL "https://github.com/go-task/task/releases/download/v3.7.0/task_linux_amd64.tar.gz" | tar xz -C "$TOOL_DEST" task
+# Install Task
+write-verbose "Checking for go-task"
+if [ "$FORCE" == true ] || [ ! -f "$TOOL_DEST/task" ]; then 
```

Comment:
a helper like `should_install "$TOOL_DEST/xyz"` might be helpful

Reply:
Good idea - makes things more declarative in a good way.